src/green/Green.jl

```diff
@@ -38,25 +47,173 @@ mutable struct Green2{T<:AbstractFloat,TGT,SGT,CT}
-Base.getindex(green::Green2, i) = green.value[i]
-#Base.lastindex(green::GreenUR) = grid.size
+"""
+Green's function with two external legs that has in-built Discrete Lehmann Representation.
+
+#Members:
+- 'timeType': Whether the Green's function is in time/frequency/dlr space
+- 'spaceType': Whether the Green's function is in coordinate space/momentum space
+- 'isFermi': Particle is fermi or boson
+- 'timeSymmetry': Whether the Green's function has particle-hole symmetry, anti-particle-hole symmetry or none of them
+- 'spaceSymmetry': Symmetry of lattice
+- 'β': Inverse temperature
+- 'Euv': the UV energy scale of the spectral density
+- 'rtol': tolerance absolute error
+- 'color': Indices of species of Green's function (such as different spin values)
+- 'timeGrid': Time or Frequency grid
+- 'spaceGrid': Coordinate or momentum grid
+- 'instant': Instantaneous part of Green's function that is proportional to δ(τ) in τ space.
+- 'dynamic': Dynamic part of Green's function
+- 'error': The error of noisy Green's function
+- 'dlrGrid': In-built Discrete Lehmann Representation
+"""
 
+mutable struct Green2DLR{T<:Number,TGT,SGT,CT}
+    timeType::Symbol
+    spaceType::Symbol
+    isFermi::Bool
+    timeSymmetry::Symbol
+    spaceSymmetry::Any
+    β::Float64
+    Euv::Float64
+    rtol::Float64
+    color::CT
+    timeGrid::TGT
+    spaceGrid::SGT
+    instant::AbstractArray{T,3}
+    dynamic::AbstractArray{T,4}
+    error::Any
+    dlrGrid::DLRGrid
 
+    """
+         function Green2DLR{T}(timeType,spaceType,isFermi, β, Euv, rtol,spaceGrid::SGT; color::CT=nothing, timeGrid::TGT=nothing,timeSymmetry=:none, spaceSymmetry=nothing,error=nothing)where{T<:Number,TGT,SGT,CT}
+    
+    Create two-leg Green's function on timeGrid, spaceGrid and color, with in-built DLR.
+    The value of instant and dynamic parts are initialized with zero.
+    #Arguements
+    - 'timeType': Whether the Green's function is in time/frequency/dlr space
+    - 'spaceType': Whether the Green's function is in coordinate space/momentum space
+    - 'isFermi': Particle is fermi or boson
+    - 'timeSymmetry': Whether the Green's function has particle-hole symmetry, anti-particle-hole symmetry or none of them
+    - 'spaceSymmetry': Symmetry of lattice
+    - 'β': Inverse temperature
+    - 'Euv': the UV energy scale of the spectral density
+    - 'rtol': tolerance absolute error
+    - 'color': Indices of species of Green's function (such as different spin values). Default: One element array [1]
+    - 'timeGrid': Time or Frequency grid. Default: DLR grid in timeType (frequency, tau or dlr) 
+    - 'spaceGrid': Coordinate or momentum grid
+    - 'error': The error of noisy Green's function
+    - 'dlrGrid': In-built Discrete Lehmann Representation
+    """
+    function Green2DLR{T}(timeType,spaceType,isFermi, β, Euv, rtol,spaceGrid::SGT; color::CT=nothing, timeGrid::TGT=nothing,timeSymmetry=:none, spaceSymmetry=nothing,error=nothing)where{T<:Number,TGT,SGT,CT}
```

Comment:
reorder the parameters, so that the related parameters stay close with each other